src/ray/util/container_util.h

```diff
@@ -97,19 +97,33 @@ typename C::mapped_type &map_find_or_die(C &c, const typename C::key_type &k) {
       map_find_or_die(const_cast<const C &>(c), k));
 }
 
-/// Remove elements whole matcher returns true against the element.
-///
-/// @param matcher the matcher function to be applied to each elements
-/// @param container the container of the elements
-template <typename T>
-void remove_elements(std::function<bool(T)> matcher, std::deque<T> &container) {
-  auto itr = container.begin();
-  while (itr != container.end()) {
-    if (matcher(*itr)) {
-      itr = container.erase(itr);
+template <typename K, typename V>
+void erase_if(absl::flat_hash_map<K, std::deque<V>> &map,
```

Comment:
[nit] do we want to return a size_t of removed elements?
[nit] Add comment on guarantees: each element is applied exactly once, in order